src/api/ProcessesApi.ts

```diff
@@ -0,0 +1,207 @@
+import { spWebContext } from "../providers/SPWebContext";
+import { IPerson, IProcess, ParentOrgs, ProcessTypes, SetAsideRecommendations, Stages } from "./DomainObjects";
+import "@pnp/sp/folders";
+import "@pnp/sp/files/folder";
+import { DateTime } from "luxon";
+import "@pnp/sp/content-types/list";
+import { sp } from "@pnp/sp";
+
+declare var _spPageContextInfo: any;
+
+export interface IProcessesApi {
+    fetchProcesses(): Promise<IProcess[]>,
+    submitProcess(process: IProcess): Promise<IProcess>,
+    deleteProcess(processId: number): Promise<void>
+}
+
+export default class ProcessesApi implements IProcessesApi {
+
+    processesList = spWebContext.lists.getByTitle("Processes");
+    dd2579ContentTypeId: string | undefined;
+    ispContentTypeId: string | undefined;
+
+    async fetchProcesses(): Promise<IProcess[]> {
+        const processes: SPProcess[] = await this.processesList.items.select("Id", "ProcessType", "SolicitationNumber", "ProgramName", "ParentOrg", "Org", "Buyer/Id", "Buyer/Title", "Buyer/EMail", "ContractingOfficer/Id", "ContractingOfficer/Title", "ContractingOfficer/EMail", "SmallBusinessProfessional/Id", "SmallBusinessProfessional/Title", "SmallBusinessProfessional/EMail", "SboDuration", "ContractValueDollars", "SetAsideRecommendation", "MultipleAward", "Created", "Modified", "CurrentStage", "CurrentAssignee/Id", "CurrentAssignee/Title", "CurrentAssignee/EMail", "SBAPCR/Id", "SBAPCR/Title", "SBAPCR/EMail", "BuyerReviewStartDate", "BuyerReviewEndDate", "COInitialReviewStartDate", "COInitialReviewEndDate", "SBPReviewStartDate", "SBPReviewEndDate", "SBAPCRReviewStartDate", "SBAPCRReviewEndDate", "COFinalReviewStartDate", "COFinalReviewEndDate").expand("Buyer", "ContractingOfficer", "SmallBusinessProfessional", "CurrentAssignee", "SBAPCR").filter("IsDeleted ne 1 and (ProcessType eq 'DD2579' or ProcessType eq 'ISP')").get();
+        return processes.map(p => this.spProcessToIProcess(p));
+    }
```

Comment:
I'm late to the party, but did you want to try the PnPJS Entity Mapping?
https://pnp.github.io/pnpjs/sp/entity-merging/
Not sure what other side effects we may see...